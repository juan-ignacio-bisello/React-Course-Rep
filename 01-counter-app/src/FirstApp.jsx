import PropTypes from 'prop-types';

export const FirstApp = ( { title, subTitle } ) => {


    const newMessage = 'Ignacio';
    return ( 
        // <> funciona como un Fracmento sin la necesidad de hacer la importaci√≥n
        <>
            <h1>{ title }</h1>

            <p>{ subTitle }</p>
            
            <p>No se </p>
        </>
        
    );
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
    title:'No hay',
    subTitle:'No hay subtitulo'
}