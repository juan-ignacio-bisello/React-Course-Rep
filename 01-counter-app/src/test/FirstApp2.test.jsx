/** * @jest-environment jsdom */

import { getByTestId, render, screen } from "@testing-library/react"
import { FirstApp } from "../FirstApp"


describe('Puebas en FirstApp2', () => {
    
    const title = 'Hola soy goku';

    test('Snapshot', () => { 
      
      const { container } = render(<FirstApp title={ title}/>);
      expect( container ).toMatchSnapshot();
     });

    test('Debe mostrar "Hola soy goku"', () => { 
      
      render(<FirstApp title={ title} /> );
      
      // screen.debug();

      expect( screen.getByText(title) ).toBeTruthy();
    });

    test('Debe mostrar el titulo en un h1', () => { 
      
      render(<FirstApp title={ title} /> );

      expect( screen.getByRole('heading', { level: 1}).innerHTML ).toContain( title );
     });

    test('debe de mostrar el subtitulo enviado por props', () => {
        
      render( 
          <FirstApp 
              title={ title }
              subTitle={ subTitle }
          />  
      );

      expect( screen.getAllByText(subTitle).length ).toBe(2);

    });
    
})

