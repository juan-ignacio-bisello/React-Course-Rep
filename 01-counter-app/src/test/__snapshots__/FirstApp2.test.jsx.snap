// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en FirstApp2 Snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola soy goku
     
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No se 
  </p>
</div>
`;
